packages/next/src/server/base-server.ts

```diff
@@ -1183,18 +1182,21 @@ export default abstract class Server<
           const originQueryParams = { ...parsedUrl.query }
 
           const pathnameBeforeRewrite = parsedUrl.pathname
-          const rewriteParamKeys = Object.keys(
-            utils.handleRewrites(req, parsedUrl)
+          const { rewriteParams, rewrittenParsedUrl } = utils.handleRewrites(
+            req,
+            parsedUrl
           )
+          const rewriteParamKeys = Object.keys(rewriteParams)
 
           // Create a copy of the query params to avoid mutating the original
           // object. This prevents any overlapping query params that have the
           // same normalized key from causing issues.
-          const queryParams = { ...parsedUrl.query }
-          const didRewrite = pathnameBeforeRewrite !== parsedUrl.pathname
+          const queryParams = { ...rewrittenParsedUrl.query }
```

Comment:
```suggestion
          const queryWithRewriteParams = { ...rewrittenParsedUrl.query }
```
Maybe signal why these query params are different.